dnf/base.py

```diff
@@ -1744,9 +1753,111 @@ def install_module(self, specs, strict=True):
+        # when enabling different stream for kickstart
+        former_assumeyes = self.conf.assumeyes
+        if not self.output:
+            self.conf.assumeyes = True
+        groups = self.install_module(module_specs)
+        self.conf.assumeyes = former_assumeyes
+        return groups
+
+    def exclude_groups(self, group_specs):
+        group_excludes = []
+
+        for group_spec in group_specs:
+            if '/' in group_spec:
+                split = group_spec.split('/')
+                group_spec = split[0]
+
+            if group_spec[0] == "^":
+                environment = self.comps.environment_by_pattern(group_spec[1:])
+                if not environment:
+                    continue
+
+            for group in environment.groups_iter():
+                for pkg in group.packages_iter():
+                    group_excludes.append(pkg.name)
+
+        exclude_query = self.sack.query().filter(name=group_excludes)
+        self.sack.add_excludes(exclude_query)
+
+    def install_groups(self, group_specs, exclude_packages, ignored):
+        for group_spec in group_specs:
+            try:
+                types = self.conf.group_package_types
+
+                if '/' in group_spec:
+                    split = group_spec.split('/')
+                    group_spec = split[0]
+                    types = split[1].split(',')
+
+                if group_spec[0] == "^":
+                    environment = self.comps.environment_by_pattern(group_spec[1:])
+                    if environment:
+                        self.environment_install(environment.id,
+                                                 types,
+                                                 exclude=exclude_packages,
+                                                 strict=self.conf.strict)
+                    else:
+                        ignored.append("@" + group_spec)
+                else:
+                    group = self.comps.group_by_pattern(group_spec)
+
+                if group:
+                    self.group_install(group.id, types,
+                                       exclude=exclude_packages,
+                                       strict=self.conf.strict)
+                else:
+                    ignored.append("@" + group_spec)
+            except dnf.exceptions.Error:
+                ignored.append("@" + group_spec)
+
+    def install_specs(self, install, exclude=None):
+        if exclude is None:
+            exclude = []
+
+        ignored = []
+        exclude_specs, install_specs = self.classify(exclude, install)
+        self.exclude_spec(exclude_specs)
+        self.install_packages(install_specs.pkg_specs, ignored)
+        groups = self.install_modules(install_specs.grp_specs)
+
+        self.read_comps(arch_filter=True)
+        self.exclude_groups(exclude_specs.grp_specs)
+        self.install_groups(groups, exclude_specs.pkg_specs, ignored)
+
+        return ignored
```

Comment:
Returning skipped packages or objects is against philosophy of DNF. Please adopt a behavior of other base.install functions with aspect of strict behavior.

Reply:
Requirement from Anaconda.